todo_app/templates/index.html

```diff
@@ -25,32 +56,29 @@ <h2>Add New Item</h2>
     </div>
 </div>
 
-<div class="p-2 mx-4 border-black-25 border-bottom"></div>
-  <div class="row">
-    <div class="col-md-12">
-      <table>
-        <tr>
-          <th>Title<form action="/sort"><button style="background-color:#535252" type="submit" class="btn btn-primary" formmethod="post" formaction="/sort" value="title" name="sortBy">Sort</button></form></th>
-          <th>Due Date</th>
-          <th>Status<form action="/sort"><button style="background-color:#535252" type="submit" class="btn btn-primary" formmethod="post" formaction="/sort" value="status" name="sortBy">Sort</button></form></th>
-          <th>Action<form action="/sort"><button style="background-color:#535252" type="submit" class="btn btn-primary" formmethod="post" formaction="/sort" value="reset" name="sortBy">Clear Sort</button></form></th>
-        </tr>
-        <!-- List the todo items here -->
-        {% for item in list %}
-            <tr>
-              <td>{{item.title}}</td>
-              <td><span {% if (item.duePast) and not ((item.status == 'Completed') or (item.status == 'Done'))%} style="background-color:#ff5959" {% endif %}>{{item.due}}</span></td>
-              <td><span {% if (item.status == 'Completed') or (item.status == 'Done') %} style="background-color:#00FF00" {% elif item.status == 'To Do' %} style="background-color:#ff5959" {% else %} style="background-color:#fcff59" {% endif %}>{{item.status}}</span></td>
-              <td><form action="/mark">
-                <button class="btn btn-warning" type="submit" formmethod="post" formaction="/mark" value="Doing" name="item_state" >In Progress</button>
-                <button class="btn btn-success" type="submit" formmethod="post" formaction="/mark" value="Done" name="item_state" >Complete</button>
-                <button class="btn btn-primary" type="submit" formmethod="post" formaction="/mark" value="To Do" name="item_state" >To Do</button>
-                <button class="btn btn-danger" type="submit" formmethod="post" formaction="/rem" value="Delete" name="item_state">Remove</button>
-                <input type="hidden" name="item_id" value="{{item.id}}"><br><br></td>
-              </form>
-            </tr>
-        {% endfor %} 
-      </table><br>
-    </div>
-  </div>
-{% endblock %}
+<!-- IF statements are used here to determine which tables to show - this reduces clutter on screen -->
+<!-- Only display the ToDo table if we have items to display -->
+  {% if viewModel.todo_items|count > 0 %}
+    {{ display_table('To Do', viewModel.todo_items, 'todo') }}
+  {% endif %}
+
+<!-- Only display the Doing table if we have items to display-->
+  {% if viewModel.doing_items|count > 0 %}
+    {{ display_table('Doing', viewModel.doing_items, 'doing') }}
+  {% endif %}
+
+<!-- Only display all the done items if we have 3 or less -->
```

Comment:
Note that this comment is wrong in exactly the sort of way comments often do get out of date! The `should_show_all_done_items` name gives developers a clear place to investigate if they need to care why it's showing/not showing, but hides the details to let them change safely if needed